kernel/mem.c

```
@@ -1,161 +1,426 @@
+
+void zero_bss() {
+  uint8_t *p = &__bss_start__;
+  while (p < &__bss_end__) {
+    *p++ = 0;
+  }
+  heap_ptr = &__bss_end__;
+}
+
+char *strdup(const char *s) {
+  char *a = malloc(strlen(s) + 1);
+  memcpy(a, s, strlen(s) + 1);
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)